hardware_interface/include/hardware_interface/components/base_interface.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__COMPONENTS__BASE_INTERFACE_HPP_
+#define HARDWARE_INTERFACE__COMPONENTS__BASE_INTERFACE_HPP_
+
+#include <string>
+
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/types/hardware_interface_status_values.hpp"
+
+namespace hardware_interface
+{
+namespace components
+{
+
+template<class InterfaceType>
+class BaseInterface : public InterfaceType
+{
+public:
+  return_type configure(const HardwareInfo & info) override
+  {
+    info_ = info;
+    status_ = status::CONFIGURED;
+    return return_type::OK;
+  }
+
+  return_type configure_default(const HardwareInfo & info)
+  {
+    return BaseInterface<InterfaceType>::configure(info);
+  }
+
+  std::string get_name() const final
+  {
+    return info_.name;
+  }
+
+  status get_status() const final
+  {
+    return status_;
+  }
+
+protected:
+  HardwareInfo info_;
```

Comment:
Should I rename `info_` to `hardware_info_`? This is used in all child classes so that it could be more explicit for users.

Reply:
nah